src/backend/computeFunction.js

```diff
@@ -0,0 +1,34 @@
+//@flow
+
+import {type WeightedGraph} from "../core/weightedGraph";
+import {TaskReporter} from "../util/taskReporter";
+import {type TimelineCredParameters} from "../analysis/timeline/params";
+import {type PluginDeclaration} from "../analysis/pluginDeclaration";
+import {TimelineCred} from "../analysis/timeline/timelineCred";
+
+/**
+ * An abstract handle for TimelineCred.compute-like functions.
+ */
+export type ComputeFunction = (opts: ComputeOpts) => Promise<TimelineCred>;
+
+// Note: type should not be strict, it's expected to be a subset.
+type ComputeEnv = {
+  +reporter: TaskReporter,
+};
+
+type ComputeOpts = {|
+  weightedGraph: WeightedGraph,
+  params?: $Shape<TimelineCredParameters>,
+  plugins: $ReadOnlyArray<PluginDeclaration>,
```

Comment:
You can put
`// TODO(@decentralion, #1557): remove plugins arg`
here if you want

Reply:
Added this.